### PR TITLE
feat(api): introduce locator.waitFor

### LINKED ISSUE #8577: [Feature] Extend locator api with functions like waitForSelector()
Given the pseudo-POM below and the case where you would need to wait for an element for any reason, you'd have to repeat the selector in two places. It hurts reusability.

```
export class PageObject {
  private readonly page: Page
  private readonly someButton: Locator
  
  constructor(page: Page) {
    this.page = page
    this.someButton = this.page.locator('#someButtton')
  }
  
  async clickSomeButton() {
    await this.page.waitForSelector('#someButton')
    await this.someButton.click()
  }
}
```

While using the old method, you could set selector value once and then simply reuse the variable as much as you want:
<details>
  <summary>POM with page actions</summary>

```
export class PageObject {
  private readonly page: Page
  private readonly someButton: string

  constructor(page: Page) {
    this.page = page
    this.someButton = ('#someButtton')
  }

  async clickSomeButton() {
    await this.page.waitForSelector(this.someButton)
    await this.page.click(this.someButton)
  }
}
```
</details>

It'd be nice if locator API would have an option to use functions like waitForSelector(). My use case involves only this function but there might be others.

### PR DESCRIPTION
**Pull request by pavelfeldman**
Fixes https://github.com/microsoft/playwright/issues/8577

### CODE REVIEW

File: src/client/locator.ts [hunk 1]
```diff
@@ -213,6 +213,14 @@ export class Locator implements api.Locator {
     return this._frame.$$eval(this._selector, ee => ee.map(e => e.textContent || ''));
   }
 
+  waitFor(options: channels.FrameWaitForSelectorOptions & { state: 'attached' | 'visible' }): Promise<void>;
+  waitFor(options?: channels.FrameWaitForSelectorOptions): Promise<void>;
+  async waitFor(options?: channels.FrameWaitForSelectorOptions): Promise<void> {
+    return this._frame._wrapApiCall(async (channel: channels.FrameChannel) => {
+      await channel.waitForSelector({ selector: this._selector, strict: true, ...options });
```
Comment (dgozman) [inline src/client/locator.ts:220]: Forgot to pass `omitReturnValue` here?

File: tests/page/locator-misc-2.spec.ts [hunk 2]
```diff
@@ -82,3 +82,12 @@ it('should return bounding box', async ({ page, server, browserName, headless, i
   const box = await element.boundingBox();
   expect(box).toEqual({ x: 100, y: 50, width: 50, height: 50 });
 });
+
+it('should waitFor', async ({ page }) => {
```
Comment (dgozman) [inline tests/page/locator-misc-2.spec.ts:86]: Let's add a test with `waitFor({ hidden })` or some other state passed, otherwise we lack coverage.

**Review by dgozman — APPROVED** [review]

**Comment by sarweshkumar2018** [comment]
How to handle if the locator returns more than one elements. 
await this.page.locator('[aria-label="bene"]').waitFor(); 

await this.page.locator('[aria-label="bene"]:first-child').waitFor() --- is NOT working

**Comment by xyide** [comment]
> How to handle if the locator returns more than one elements.

I don't think it would be necessary to handle this case.  If any element with a given selector us visible, then naturally the first-child is visible.

**Pull request merged**